feat: make multiselect menu more discoverable

GretaD opened the pull request:
fixes #10231

hovered
![Screenshot from 2024-12-09 14-47-01](https://github.com/user-attachments/assets/3793a4c7-4ec8-49d3-a688-48b312d80d54)

selected
![Screenshot from 2024-12-09 14-49-25](https://github.com/user-attachments/assets/cbe54311-a3e1-4a30-91d2-ae622bc3ce7e)



To clarify
- [x] Should the checkmark be shown when the whole message is hovered, or when the avatar is hovered. Right now, the check mark is shown when the avatar is hovered.

GVodyanov reviewed (commented):
![image](https://github.com/user-attachments/assets/2ed16a4c-34c6-4c50-a048-3b5dedee9ce4)

On dark mode the select icon is the wrong color

GretaD commented:
> ![image](https://private-user-images.githubusercontent.com/76647797/393879044-2ed16a4c-34c6-4c50-a048-3b5dedee9ce4.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzM3NTU4ODgsIm5iZiI6MTczMzc1NTU4OCwicGF0aCI6Ii83NjY0Nzc5Ny8zOTM4NzkwNDQtMmVkMTZhNGMtMzRjNi00YzUwLWEwNDgtM2I1ZGVkZWU5Y2U0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMDklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjA5VDE0NDYyOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQ3OTIxMzBmYTdlMGVlNTkwZDk2NTI4YTQxOTM4ODM4N2UyOGQ0ZTA1ZWFkZjBjN2Q4M2Y3YTkzMmJjYzQ0MmYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.I-Ximf0w-QIzS2ZmxrOfjr2M7FsHi3WPLWXA10JYsDs)
> 
> On dark mode the select icon is the wrong color

oh its on both modes actually, sorry, will fix soon

ChristophWurst commented on src/components/EnvelopeList.vue:
```diff
@@ -674,4 +568,7 @@ div {
 		display: block;
 	}
 }
+.action-buttons {
```
this is a child element of `.multiselect-header`. Move it in there to get into a BEM-like structure?

ChristophWurst commented on src/components/EnvelopeList.vue:
```diff
@@ -6,208 +6,91 @@
-						:close-after-click="true"
+				<div class="action-buttons">
+					<NcButton type="tertiary"
+						:title="areAllSelectedRead ? n('mail', 'Mark {number} unread', 'Mark {number} unread', selection.length, { number: selection.length }) : n('mail', 'Mark {number} read', 'Mark {number} read', selection.length, { number: selection.length })"
```
nit: the previous multi-line wrap made the `n` function call more readable than this one-liner

ChristophWurst commented:
* If all selected messages are read, there is a bulk *Mark us unread*. If at least one message is unread there is only a bulk *Mark as read* and no more *Mark as unread*.
* There is no option to un-flag (un-star) a message (was not there before either)
* Before it was possible to clear the selection. This feature is gone.
* The expansion of the multiselect buttons makes the list move and at least on Firefox it's a bit jittery.
* The hover state of the avatar of a selected message is identical to a message that is not selected. This means you get no click feedback when selecting a message until you leave the element again.

![image](https://github.com/user-attachments/assets/75626d2e-4857-4a44-bd5a-70856fa448a7)

![image](https://github.com/user-attachments/assets/373d2b20-08b8-4317-a826-1298e7782e9b)

GretaD commented:
> * If all selected messages are read, there is a bulk _Mark us unread_. If at least one message is unread there is only a bulk _Mark as read_ and no more _Mark as unread_.

Same scenario as favorite, unfavorite, explained below

> * There is no option to un-flag (un-star) a message (was not there before either)

It is there, i added it, but its one button that replaces each other. I just changed the icon for both states.
![Screenshot from 2024-12-10 12-56-06](https://github.com/user-attachments/assets/af76b0a2-a71e-41e6-a331-fc4d6ebff00b)
![Screenshot from 2024-12-10 12-54-57](https://github.com/user-attachments/assets/fb1e3268-dd22-4c7f-a0fe-056ba1c93502)

> * Before it was possible to clear the selection. This feature is gone.

You mean _unselect_, right? Its the second icon before last.

> * The expansion of the multiselect buttons makes the list move and at least on Firefox it's a bit jittery.

I'll test it there and fix it.

> * The hover state of the avatar of a selected message is identical to a message that is not selected. This means you get no click feedback when selecting a message until you leave the element again.
> 
Yes, you need to select the avatar to actually select it. Selecting the message will open the message and should not select it( :heavy_check_mark:  as an action). These 2 actions should be split, you shouldnt select :heavy_check_mark: a message when you only want to open it. So, i would say the hover state of the avatar should be the same, regardless of the state of the message, but, i can change it if you have an idea how.

I havent changed anything from the logic that was there. The behavior is the same as the original menu actions, i only added new icons. So, you propose to have both buttons out like in the _mark as important_ example? When you have both actions selected, like for example, one favorite, and one not favorite, to show both icons, favorite and unfavorite?
In general, you dont expect people to do that.

> ![image](https://private-user-images.githubusercontent.com/1374172/394216150-75626d2e-4857-4a44-bd5a-70856fa448a7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzM4MzAxNTgsIm5iZiI6MTczMzgyOTg1OCwicGF0aCI6Ii8xMzc0MTcyLzM5NDIxNjE1MC03NTYyNmQyZS00ODU3LTRhNDQtYmQ1YS03MDg1NmZhNDQ4YTcucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTIxMCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDEyMTBUMTEyNDE4WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZWE0MDgzN2Q0YjgyYWMyYTYyNGI2YjQwNWJkZDNmODg4NWVjOTIyNTliZjE2MTEyYmRiOWRiMDI0ZmFhZTQzZiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.duPc0Ed78WjdQhB85T5hin_d41Phejpj4k-EB8YfdnQ)

ChristophWurst commented:
> When you have both actions selected, like for example, one favorite, and one not favorite, to show both icons, favorite and unfavorite?
> In general, you dont expect people to do that.

Yes. For *Mark important/unimportant* we also have both options

SebastianKrupinski reviewed (approved):
Tested. Looks and works great. Much better visually, I think.

![image](https://github.com/user-attachments/assets/a86254ca-25a8-4b3b-86d6-d723a27aa6f1)
